components/Switch.tsx

```diff
@@ -0,0 +1,401 @@
+    z.infer<typeof formSchemaSwitch>
+  >({
+    resolver: zodResolver(formSchemaSwitch),
+    defaultValues: {
+      source: filteredLocations.find((v) => v.id === "assethub"),
+      // to do: handle this better
+      destination: initialDestination ?? "",
+      token: initialDestination?.erc20tokensReceivable[0].id,
+      beneficiary: polkadotAccount?.address ?? "",
+      sourceAccount: polkadotAccount?.address ?? "",
+      amount: "0.0",
+    },
+  });
+
+  const source = form.watch("source");
+  const destination = form.watch("destination");
+  const sourceAccount = form.watch("sourceAccount");
+  const beneficiary = form.watch("beneficiary");
+  const amount = form.watch("amount");
+  const token = form.watch("token");
+
+  const beneficiaries: AccountInfo[] = useMemo(
+    () =>
+      polkadotAccounts?.map((x) => ({
+        key: x.address,
+        name: x.name || "",
+        type: destination.type,
+      })) || [],
+    [polkadotAccounts, destination.type],
+  );
+
+  const handleTransaction = useCallback(async () => {
+    if (
+      !context ||
+      !beneficiary ||
+      !source ||
+      !destination ||
+      !sourceAccount ||
+      !token
+    ) {
+      return;
+    }
+
+    const amountInSmallestUnit = parseUnits(amount, source.paraInfo?.decimals);
+    if (!amountInSmallestUnit) {
+      return;
+    }
+    const sendTransaction = async (
+      transaction: SubmittableExtrinsic<"promise", ISubmittableResult>,
+      transactionFee: string,
+    ) => {
+      setTransaction(transaction);
+      setFeeDisplay(transactionFee);
+    };
+
+    if (source.id === "assethub") {
+      if (destination.id === "assethub") {
+        return;
+      }
+      await submitAssetHubToParachainTransfer({
+        context,
+        beneficiary,
+        source,
+        destination,
+        amount: amountInSmallestUnit,
+        sourceAccount,
+        setError,
+        setBusyMessage,
+        sendTransaction,
+      });
+    } else {
+      const { pallet } = parachainConfigs[source.name];
+
+      submitParachainToAssetHubTransfer({
+        context,
+        beneficiary,
+        source,
+        amount: amountInSmallestUnit,
+        pallet,
+        sourceAccount,
+        setError,
+        setBusyMessage,
+        sendTransaction,
+      });
+    }
+  }, [context, beneficiary, source, destination, sourceAccount, token, amount]);
+
+  useEffect(() => {
+    handleTransaction();
+  }, [handleTransaction]);
+
+  const handleSufficientTokens = (result: boolean) => {
+    setSufficientTokenAvailable(result);
+  };
+  const handleTopUpCheck = (result: boolean) => {
+    setTopUpCheck(result);
+  };
+  const onSubmit = useCallback(async () => {
+    if (!transaction || !context) {
+      return;
+    }
+
+    // to do: better error information for the user.
+    try {
+      if (destination.id === "assethub" && !sufficientTokenAvailable) {
+        setError({
+          title: "Not Enough Sufficient tokens",
```

Comment:
```suggestion
          title: "Inufficient tokens",
```

Reply:
Or "Existential deposit required" if with 'sufficient' you meant that you need to hold a 'sufficient' asset

Reply:
better? https://github.com/Snowfork/snowbridge-app/pull/23/commits/9cd1da27c669c03186fe6a87466519c94f98469a